frontend/src/hooks/queryBuilder/useFetchKeysAndValues.ts

```diff
@@ -54,49 +76,56 @@ export const useFetchKeysAndValues = (
 	const handleFetchOption = async (
 		value: string,
 		query: IBuilderQuery,
+		keys: BaseAutocompleteData[],
 	): Promise<void> => {
-		if (value) {
-			// separate the search value into the attribute key and the operator
-			const [tKey, operator] = separateSearchValue(value);
-			setResults([]);
-			if (tKey && operator) {
-				const { payload } = await getAttributesValues({
-					searchText: searchValue,
-					dataSource: query.dataSource,
-					aggregateOperator: query.aggregateOperator,
-					aggregateAttribute: query.aggregateAttribute.key,
-					attributeKey: tKey,
-				});
-				if (payload) {
-					const values = Object.values(payload).find((el) => !!el);
-					if (values) {
-						setResults(values);
-					} else {
-						setResults([]);
-					}
-				}
-			}
+		if (!value) {
+			return;
+		}
+		const { tagKey, tagOperator, tagValue } = getTagToken(value);
```

Comment:
Since this is using the `TAG_FSM`, there is no need to mandate the user to add space, or there is no need for us to add some custom logic to add space.